src/components/menu/header-messages.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import { removeHeaderMessage } from '../../actions/actions'
+import { dispatch } from '../../store'
+
+export class HeaderMessagesUnconnected extends React.Component {
+  static propTypes = {
+    messages: PropTypes.arrayOf(PropTypes.shape({
+      message: PropTypes.string,
+      button: PropTypes.string,
+      callback: PropTypes.function,
+      id: PropTypes.number,
+    })),
+  }
+  render() {
+    const handler = (id, callback) => () => {
+      if (callback()) {
+        dispatch(removeHeaderMessage(id))
+      }
+    }
+
+    const closeHandler = id => () => dispatch(removeHeaderMessage(id))
+
+    const items = this.props.messages.map(({
+      message,
+      button,
+      id,
+      callback,
+    }) => (
+      <div>
```

Comment:
I'm getting

```
Warning: Each child in an array or iterator should have a unique "key" prop.
```
with this.  I'll have to sort that out.

Reply:
this one is easy, try:
```jsx
<div key={`item-${id}`}>`
```